typings/embedded_network/websocket/client.d.ts

```diff
@@ -0,0 +1,56 @@
+declare module "embedded:network/websocket/client" {
+	import type { DNSUDPDevice } from "embedded:network/dns/resolver/udp";
+	import type { Buffer } from "embedded:io/_common";
+	import type TCP from "embedded:io/socket/tcp";
+	import type { TCPDevice } from "embedded:io/socket/tcp";
+	import type TLSSocket from "embedded:io/socket/tcp/tls";
+	import type { TLSDevice } from "embedded:io/socket/tcp/tls";
+	import "embedded:network/websocket/client-device";
+	
+	interface WebSocketClientReadableOptions {
+		more: boolean;
+		binary: boolean;
+	}
+
+	export interface WebSocketClientWriteOptions {
+		binary?: boolean;
+		more?: boolean;
+		opcode?: WebSocketClientOpcode;
+	}
+
+	type WebSocketClientOpcode = 1 | 2 | 8 | 9 | 10;
+
+	type WebSocketClientOptions = ((
+		{
+			attach?: TCP | TLSSocket;
+		} | {
+			host?: string;
+			port?: number;
+			socket: TCPDevice | TLSDevice;
+		}) & {
+			protocol?: string;
+			headers?: Map<string, string | string[]>;
+			dns?: DNSUDPDevice;
+			onReadable?: (this: WebSocketClient, count: number, options: WebSocketClientReadableOptions) => void;
+			onWritable?: (this: WebSocketClient, count: number) => void;
+			onControl?: (this: WebSocketClient, opcode: WebSocketClientOpcode, buffer: Uint8Array) => void; // should this be ArrayBuffer?
```

Comment:
Yes, `ArrayBuffer` is correct as `onControl` is [specified](https://419.ecma-international.org/#-24-websocket-client-class-pattern-data-format).

Reply:
Oops.  That was an old comment that I forgot to remove.